### PR TITLE
feature/IESP-143

### PR DESCRIPTION
**Pull request by melissa-dellorco**
Deleted RestClient and added FeignClient
Fix bug for deleting queued message after failed save

### CODE REVIEW

File: src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java [hunk 1]
```diff
@@ -37,36 +38,54 @@ public static ServicesReceiver getInstance() throws IOException {
...
-          EserviceDTO service = mapper.readValue(message.getBody(), EserviceDTO.class);
-          RestClient.getInstance().saveEservice(service, client);
-          logger.logMessageEserviceSaved(service.getEserviceId(), service.getVersionId());
-          sqs.getAmazonSQSAsync().deleteMessage(
-              new DeleteMessageRequest().withQueueUrl(sqsUrlServices)
-                  .withReceiptHandle(message.getReceiptHandle()));
-          logger.logMessageQueueMessageDeleted(service.getEserviceId(), service.getVersionId(),
-              URI.create(sqsUrlServices));
+
+    ObjectMapper mapper = new ObjectMapper();
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java:53]: Use ObjectMapper as a Singleton and not as a prototype, centralized configuration of it.

File: src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java [hunk 2]
```diff
@@ -37,36 +38,54 @@ public static ServicesReceiver getInstance() throws IOException {
...
+    SqsConfig sqs = SqsConfig.getInstance();
+
+    ReceiveMessageRequest receiveMessageRequest =
+        new ReceiveMessageRequest(sqsUrlServices).withMaxNumberOfMessages(10);
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java:57]: Use a static Constructor of , in lombok you can find one.
` @RequiredArgsConstructor(staticName = "of")`

File: src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java [hunk 3]
```diff
@@ -37,36 +38,54 @@ public static ServicesReceiver getInstance() throws IOException {
...
+    List<Message> sqsMessages =
+        sqs.getAmazonSQSAsync().receiveMessage(receiveMessageRequest).getMessages();
+
+    while (Objects.nonNull(sqsMessages) && !sqsMessages.isEmpty()) {
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java:61]: AWS client  receiveMessage return a ReceiveMessageResult with a null list, are you sure? Maybe an empty list.

File: src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java [hunk 4]
```diff
@@ -37,36 +38,54 @@ public static ServicesReceiver getInstance() throws IOException {
...
+      for (Message message : sqsMessages) {
+        EserviceDTO service = mapper.readValue(message.getBody(), EserviceDTO.class);
+
+        EserviceClient eserviceClient = Feign.builder().client(new OkHttpClient())
+            .encoder(new GsonEncoder()).decoder(new GsonDecoder())
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java:66]: Maybe is it possible to centralized its configuration as a Singleton?

File: src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java (hunk 2 above)
Comment (nttdata-rtorsoli) [reply src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java:57]: I am sorry. It is an AWS class, then if it has not a static constructor or builder you can leave so.

File: src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java (hunk 3 above)
Comment (melissa-dellorco) [reply src/main/java/it/pagopa/interop/probing/eservice/registry/updater/consumer/ServicesReceiver.java]: done

**Review by galales — APPROVED** [review]

**Pull request merged**